pkg/asset-manager-utils/test/SinglePoolAaveATokenAssetManager.test.ts

```diff
@@ -0,0 +1,330 @@
+import { ethers } from 'hardhat';
+import { Contract } from 'ethers';
+import { expect } from 'chai';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+
+import TokenList from '@balancer-labs/v2-helpers/src/models/tokens/TokenList';
+
+import { bn, fp } from '@balancer-labs/v2-helpers/src/numbers';
+import { MAX_UINT256 } from '@balancer-labs/v2-helpers/src/constants';
+
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import Vault from '@balancer-labs/v2-helpers/src/models/vault/Vault';
+import { expectBalanceChange } from '@balancer-labs/v2-helpers/src/test/tokenBalance';
+import { advanceTime } from '@balancer-labs/v2-helpers/src/time';
+import { encodeJoinWeightedPool } from '@balancer-labs/v2-helpers/src/models/pools/weighted/encoding';
+
+const OVER_INVESTMENT_REVERT_REASON = 'investment amount exceeds target';
+const UNDER_INVESTMENT_REVERT_REASON = 'withdrawal leaves insufficient balance invested';
+
+const tokenInitialBalance = bn(200e18);
+const amount = bn(100e18);
+
+const setup = async () => {
+  const [, admin, lp, other] = await ethers.getSigners();
+
+  const tokens = await TokenList.create(['DAI', 'MKR'], { sorted: true });
+
+  // Deploy Balancer Vault
+  const vaultHelper = await Vault.create({ admin });
+  const vault = vaultHelper.instance;
+
+  // Deploy mocked Aave
+  const lendingPool = await deploy('MockAaveLendingPool', { args: [] });
+  const aaveRewardsController = await deploy('MockAaveRewards');
+  const stkAave = aaveRewardsController;
+
+  const daiAToken = await deploy('MockAToken', { args: [lendingPool.address, 'aDai', 'aDai', 18] });
+  await lendingPool.registerAToken(tokens.DAI.address, daiAToken.address);
+
+  // Deploy Asset manager
+  const assetManager = await deploy('SinglePoolAaveATokenAssetManager', {
+    args: [
+      vault.address,
+      tokens.DAI.address,
+      lendingPool.address,
+      daiAToken.address,
+      aaveRewardsController.address,
+      stkAave.address,
+    ],
+  });
+  const assetManagers = [assetManager.address, other.address];
+
+  // Deploy Pool
+  const args = [
+    vault.address,
+    'Test Pool',
+    'TEST',
+    tokens.addresses,
+    [fp(0.5), fp(0.5)],
+    assetManagers,
+    fp(0.0001),
+    0,
+    0,
+    admin.address,
+  ];
+
+  const pool = await deploy('v2-pool-weighted/WeightedPool', {
+    args,
+  });
```

Comment:
Using the WeightedPool model instead of calling the contracts directly will not only greatly simplify this, but also joins, etc.

Reply:
Hmm, I saw that but I actually don't find the helper very helpful in this case.   It feels more illustrative to write these specs around a contract instance than a helper model.

Generally I think I would prefer a helper that serves more as a factory to return a contract instance than the `WeightedPool` helper which returns a decorated contract instance.  For example, bubbling all the instance methods up to the `WeightedPool` model seems like an unnecessary abstraction/source of indirection imho

Up to you.  If you insist, I'll use it.

Reply:
Yeah I know what you mean, especially regarding them possibly obscuring things. I gave this a try locally to see what the result would be, and sadly they're a little too magical at the moment and e.g. deploy their own Vault. We might improve them in the future if the need arises.